packages/ui/src/contexts/AppearanceContext.test.tsx

```diff
@@ -0,0 +1,127 @@
+import { fullTheme } from '~/themes';
+import { AppearanceProvider, useAppearance, defaultAppearance } from './AppearanceContext';
+
+describe('AppearanceContext', () => {
+  it('renders expected defaults', () => {
+    const wrapper = ({ children }: { children: ReactNode }) => <AppearanceProvider>{children}</AppearanceProvider>;
+
+    const { result } = renderHook(useAppearance, { wrapper });
+    expect(result.current).toStrictEqual({
+      parsedAppearance: {
+        layout: defaultAppearance.layout,
+        elements: defaultAppearance.elements,
+        theme: defaultAppearance.theme,
+      },
+      theme: undefined,
+      themelessAppearance: null,
+    });
+  });
+
+  it('factors in provided element styles', () => {
+    const wrapper = ({ children }: { children: ReactNode }) => (
+      <AppearanceProvider appearance={{ elements: { alert__warning: 'class-two' } }}>{children}</AppearanceProvider>
+    );
+
+    const { result } = renderHook(useAppearance, { wrapper });
+    expect(result.current).toStrictEqual({
+      parsedAppearance: {
+        layout: defaultAppearance.layout,
+        elements: {
+          ...defaultAppearance.elements,
+          alert__warning: {
+            descriptor: 'cl-alert__warning',
+            className: [defaultAppearance.elements.alert__warning.className, 'class-two'].join(' '),
+            style: {},
+          },
+        },
+        theme: defaultAppearance.theme,
+      },
+      theme: undefined,
+      themelessAppearance: {
+        elements: {
+          alert__warning: 'class-two',
+        },
+      },
+    });
+  });
+
+  it('merges multiple appearance props', () => {
+    const wrapper = ({ children }: { children: ReactNode }) => (
+      <AppearanceProvider appearance={{ elements: { alert__warning: 'class-two' } }}>
+        <AppearanceProvider appearance={{ elements: { alert__warning: 'class-three' } }}>{children}</AppearanceProvider>
+      </AppearanceProvider>
+    );
+
+    const { result } = renderHook(useAppearance, { wrapper });
+    expect(result.current).toStrictEqual({
+      parsedAppearance: {
+        layout: defaultAppearance.layout,
+        elements: {
+          ...defaultAppearance.elements,
+          alert__warning: {
+            descriptor: 'cl-alert__warning',
+            className: [defaultAppearance.elements.alert__warning.className, 'class-two', 'class-three'].join(' '),
+            style: {},
+          },
+        },
+        theme: defaultAppearance.theme,
+      },
+      theme: undefined,
+      themelessAppearance: {
+        elements: {
+          alert__warning: 'class-two class-three',
+        },
+        layout: {},
+      },
+    });
+  });
+
+  it('propogates the provided baseTheme', () => {
+    const testTheme = structuredClone(fullTheme);
+    testTheme.alert__warning.className = 'alert-test-classname';
+
+    const wrapper = ({ children }: { children: ReactNode }) => (
+      <AppearanceProvider
+        theme={testTheme}
```

Comment:
In a near-future refactor this will be moved into the `appearance` prop instead of being a prop on the provider directly.